packages/application-shell/src/components/fetch-project/fetch-project.spec.js

```diff
@@ -52,7 +52,7 @@ describe('rendering', () => {
   it('should render error state', async () => {
     mockServer.use(
       graphql.query('FetchProject', (req, res, ctx) => {
-        return res.once(ctx.status(401));
+        return res.data(ctx.status(401));
```

Comment:
@emmenko this is actually a very nice catch. The project query will fail once with 401 then result in data being loaded after.

So the fact is that the query can load twice until you get an error. I am trying to think if this is problematic but I think not. What do you think?

Reply:
Hmm let me think. Now that I see this, I'm actually wondering how is this even working...shouldn't we define the main mock in the `setupServer` as `res.once`? 🤔

Reply:
Yeah, I think that's a bit what we talked about yesterday. `res.once` is actually quite powerful. We might want to use it as the default more. While having mock handlers local is often easier to maintain than this global catch-all stuff also if it's a bit more redundant.

I tried to improve it in ac18bf5. I find such tests incredibly powerful to be honest ⭐.